crates/agent/src/api/mod.rs

```diff
@@ -234,34 +242,92 @@ where
     }
 }
 
-// Middleware which validates JWT tokens before proceeding, and attaches verified Claims.
-async fn authorize(
+// Middleware which validates a control-plane access token before proceeding,
+// and attaches verified ControlClaims encoded within its JWT.
+async fn layer_control_claims(
     axum::extract::State(app): axum::extract::State<Arc<App>>,
     axum_extra::TypedHeader(bearer): axum_extra::TypedHeader<
         axum_extra::headers::Authorization<axum_extra::headers::authorization::Bearer>,
     >,
     mut req: axum::http::Request<axum::body::Body>,
     next: axum::middleware::Next,
 ) -> axum::response::Response {
-    let token = match jsonwebtoken::decode::<ControlClaims>(
-        bearer.token(),
-        &app.control_plane_jwt_verifier,
-        &app.jwt_validation,
-    ) {
+    let claims = match verify_control_claims(app, bearer.token()) {
         Ok(claims) => claims,
         Err(err) => {
-            return (
-                StatusCode::UNAUTHORIZED,
-                format!("failed to parse authorization token: {err}"),
-            )
-                .into_response();
+            return (StatusCode::UNAUTHORIZED, format!("{err:?}")).into_response();
         }
     };
 
-    req.extensions_mut().insert(token.claims);
+    req.extensions_mut().insert(claims);
     next.run(req).await
 }
 
+// Middleware which exchanges a refresh bearer token for an access token,
+// validates the generated access token before proceeding,
+// and attaches verified ControlClaims encoded within its JWT.
+async fn layer_refresh_claims(
```

Comment:
Another possible solution to the openapi spec auth requirements might be to have a layer that accepts either an access token or a refresh token, and allow all of our routes to accept either.

Reply:
:+1: I think this is the way to go. There's no _requirement_ that his api take a refresh token, it's just not practical to configure grafana-agent / datadog-agent et all to do a token exchange flow.

I can't think of any downside to letting other APIs take a refresh token either.